packages/web-frontend/src/containers/MeasureLegend/LegendEntry.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ */
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import { connect } from 'react-redux';
+import styled from 'styled-components';
+
+const Key = styled.div`
+  padding: 5px;
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  pointer-events: auto;
+  cursor: pointer;
+  ${p => (p.hidden ? 'opacity: 0.5;' : '')}
+`;
+
+const LegendEntry = ({ marker, label, value, dataKey, onClick, hiddenMeasures, unClickable }) => {
+  const hidden = (hiddenMeasures[dataKey] || {})[value];
+
+  return (
+    <Key onClick={unClickable ? null : () => onClick(dataKey, value, !hidden)} hidden={hidden}>
+      {marker}
+      <div>{label}</div>
+    </Key>
+  );
+};
+
+const mapLegendStateToProps = () => {
+  return state => ({
+    hiddenMeasures: state.map.measureInfo.hiddenMeasures,
+  });
+};
```

Comment:
const mapStateToProps = () => state => ({
  hiddenMeasures: state.map.measureInfo.hiddenMeasures,
});

Reply:
Might as well use arrow function 'implicit return' here

Reply:
And might as well just use the conventional 'mapStateToProps'. We don't need to say legend since this is now the only mapState in the file. This is one of the benefits of having one component for file as you have so nicely done!

Reply:
Much nicer!